### PR TITLE
Add verified fake

### PR DESCRIPTION
**Pull request by imsickofmaps**

### CODE REVIEW

**Comment by imsickofmaps** [comment]
@rudigiesler - can you take a look at this? At the moment I couldn't work out how to make a fake multipart parser so I split just those tests out to be just real API tests right now

 / FYI @smn

File: contentstore/tests/tests_server.py [hunk 1]
```diff
@@ -0,0 +1,456 @@
...
+                err.args[0],
+                "(install from pypi or the 'verified_fake' directory)"]))
+
+        self.req_class = Request
+        self.api_class = FakeContentStoreApi
+        self.api = self.api_class("", "token-1", {}, {})
+        self.api.messagesets.endpoint_data = {}
+        self.api.schedules.endpoint_data = {}
+        self.api.messages.endpoint_data = {}
+        self.api.binary_contents.endpoint_data = {}
```
Comment (rudigiesler) [inline contentstore/tests/tests_server.py:237]: This seems like a lot of setup code. We probably want the user of the fake api to just be able to `api = FakeContentStoreApi(..)`. What is the reason for setting the endpoint data to empty dictionaries manually?

Comment (imsickofmaps) [inline contentstore/tests/tests_server.py:237]: if this isn't present the datastore persists between tests, this is done to force the clearing

File: contentstore/tests/tests_server.py [hunk 2]
```diff
@@ -0,0 +1,456 @@
...
+        if schedule_id is None:
+            return data.popitem()[1]
+        else:
+            return data.get(schedule_id)
+
+    def get_schedules(self):
+        data = self.api.schedules.endpoint_data
+        s = []
+        for key, schedule in data.iteritems():
+            s.append(schedule)
```
Comment (rudigiesler) [inline contentstore/tests/tests_server.py:270]: `return data.values()`? Also, is there a reason why we use `Struct` when creating to mimic Django, but just dicts when getting?

Comment (rudigiesler) [inline contentstore/tests/tests_server.py:270]: Never mind about the second comment. I see we use the serialiser for getting and return the object when creating.

**Comment by rudigiesler** [comment]
I made a few comments. I didn't really look at the tests themselves, I've just been looking at the structure of the fake and how you're running the tests against both. The structure is looking good.

File: contentstore/tests/tests_server.py (hunk 2 above)
Comment (imsickofmaps) [inline contentstore/tests/tests_server.py:270]: Sometimes you have one of those moments when you forget how to python.

File: contentstore/tests/tests_server.py (hunk 1 above)
Comment (rudigiesler) [inline contentstore/tests/tests_server.py:237]: Ah yes, this is caused by the default function values (`def __init__(self, data={})`). We could maybe do a `self.messagesets = {}`, and then create the api by `self.api = self.api_class(..., self.messagesets, ...)`. Then when creating/getting messagesets you can just do `self.messagesets[key] = value`. There's nothing wrong with doing it this way, I think it's just clearer doing it like that. I'm happy with both ways though.

**Comment by imsickofmaps** [comment]
@rudigiesler I think this is done now. I've validated it also by merging this in with the client branch #4

**Comment by rudigiesler** [comment]
@imsickofmaps Looks good. I'm assuming the tests are just copied over? If not then I should probably look through what has changed there too.

**Comment by imsickofmaps** [comment]
@rudigiesler yes - it was the helpers that changed

**Comment by rudigiesler** [comment]
In that case, :+1:

**Pull request merged**